### PR TITLE
noise injection as model perturbation 

### PR DESCRIPTION
**Pull request by yairchn**
# Earth-2 MIP Pull Request

## Description
This PR adds noise injection between time steps as a form of model perturbation. This is done by adding to `model` a noise_injection function that can be called from the Inference class, or counterpart stepping code depending on the model. This noise injection function is defined based on the config file. At the moment the noise is applied by multiplying the state vector by a factor: 
```
x = x * ( 1+amp*dt*brown_noise)
``` 
where dt is the time step of the model and the amplitude `amp` has dimension of `1/hour`. 
This allows to handle both dimensional models (Pangu) and non dimensional models (FCN).

This new noise method is different than the IC noise as it changes `x` in place, rather than returning a noise tensor. Future PR's should unify this approach and the same noise functions could be used for both IC and model noise injection. 

Closes: #51

## Checklist

- [x] I am familiar with the Contributing Guidelines.
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The CHANGELOG.md is up to date with these changes.
- [X] An [issue](https://github.com/NVIDIA/earth2mip/issues) is linked to this pull request.

### CODE REVIEW

**Comment by yairchn** [comment]
Here is how `t2m` looks for two ensemble members in Pangu 
Screenshot 2023-09-28 at 4 26 02 PM

**Comment by yairchn** [comment]
Here is how `t2m` looks for two ensemble members in fcnv2_sm
Screenshot 2023-09-28 at 3 16 12 PM

**Comment by yairchn** [comment]
This method (turning off IC perturbation) generated such trajectories

**Comment by yairchn** [comment]
Turning off IC perturbations, the model perturbations generates these trajectories 
Screenshot 2023-10-17 at 4 50 01 PM

File: earth2mip/ensemble_utils.py [hunk 1]
```diff
@@ -150,6 +150,19 @@ def to(self, *args, **kwargs):
...
+def generate_model_noise_correlated(x,
+                                    time_step,
+                                    reddening,
+                                    device,
+                                    noise_injection_amplitude,
+                                    ):
+    shape = x.shape
+    dt = torch.tensor(time_step.total_seconds()) / 3600.0
+    noise = noise_injection_amplitude * dt * brown_noise(shape, reddening).to(device)
+    return x * (1.0 + noise)
```
Comment (nbren12) [inline earth2mip/ensemble_utils.py:163]: This function can be tightened up:

```suggestion
def add_multiplicative_noise(x, amplitude, reddening):
    noise = amplitude * brown_noise(shape, reddening).to(x.device)
    return x * (1.0 + noise)
    
    
dt =  torch.tensor(time_step.total_seconds()) / 3600.0
x_perturbed = add_multiplicative_noise(x, dt * noise_injection_amplitude)
```

Comment (nbren12) [reply earth2mip/ensemble_utils.py:163]: from 5 args to 2.

**Comment by nbren12** [comment]
on more thought, I think we should merge this PR. the main blocker is that this option is model specific, but we don't have an easy way to set model specific options.

**Comment by nbren12** [comment]
@yairchn Looks #77 merged. should we close this too?

**Comment by NickGeneva** [comment]
Closing since integrated with another PR.

**Pull request closed without merging**